modules/nw-udn-cr.adoc

```diff
@@ -0,0 +1,98 @@
+//module included in the following assembly:
+//
+// *networking/multiple_networks/understanding-user-defined-networks.adoc
+
+:_mod-docs-content-type: PROCEDURE
+[id="nw-udn-cr_{context}"]
+= Creating a UserDefinedNetwork custom resource
+
+The following procedure sets up a user-defined network that is namespace scoped.
+
+.Prerequisites
+* You must
+
+.Procedure
+
+[NOTE]
+====
+If any cluster default networked pods exist before the user-defined network is created, any further pods created in this namespace will return an error message: `What_is_this`.
+====
+
+. Create a request for a user-defined network:
+
+.. Create a YAML file, such as `my-udn-request.yaml`, to define your request with content as in the following example:
++
+[source, yaml]
+----
+apiVersion: k8s.ovn.org/v1
+kind: UserDefinedNetwork
+metadata:
+  name: udn-1 # <1>
+  namespace: <some_custom_namespace>
+spec:
+  topology: Layer2 # <2>
+  layer2: <3>
+    role: Primary # <4>
+    mtu: 9000 # <5>
+    subnets:
+      - "10.0.0.0/24"
+      - "2001:db8::/64" # <6>
+    excludeSubnets:
+      - "10.0.0.0/26"
+      - "2001:db8::/80" # <7>
+    JoinSubnets: [] # <8>
+    IPAMLifecycle: Persistent # <9>
+----
+<1> Name of your `UserDefinedNetwork` resource. This should not be `default`.
+<2> The layer you want to create the network on. The `topology` field accepts values of `Layer2` and `Layer3`.
+<3> You can use this annotation to specify either `layer2` or `layer3`.
+<4> Specifies `Primary` or `Secondary`. `Primary` is the only `role` specification supported in {product-version}.
+<5> The maximum transmission units (MTU) the default value is `1400`. The boundary for IPv4 is 574, and for IPv6 it is 1280.
+<6> Specifies the subnet to be used for the network across the cluster. Supports both IPv6 and dual-stack. For example, `192.168.100.0/24`,`2001:DBB::/64`. Dual-stack may set two subnets otherwise only one is allowed. When the `topology` field is set to `Layer3`, the subnet is split into smaller subnets for every node. Accepted format for subnets when `Layer3` is set are: `172.16.0.0/16/24`. For `Layer2` values in `topology` field, standard CIDR ranges are accepted. If omitted the network only provides `Layer2` communication and you must configure IP addresses.
+<7> Specifies the CIDR range to remove from the assignable IP address pool excluding them from being assigned to pods.
+<8> Specifies the `subnets` used inside the OVN-Kubernetes network topology. If omitted, the OVN-Kubernetes network plugin assigns one, which is subject to change over time.
```

Comment:
This field is optional, and allowed when Role=Primary.

Users should know what they doing when specifying this :slightly_smiling_face: